fips/fips.go

```diff
@@ -0,0 +1,11 @@
+package fips
+
+import "os"
+
+// EnvVar is the environment variable which stores FIPS mode state
+const EnvVar = "GOFIPS"
+
+// Enabled returns true when FIPS mode is enabled
+func Enabled() bool {
+	return os.Getenv(EnvVar) != ""
```

Comment:
@cyli is it sufficient to just check this?

Reply:
@nishanttotla The requirements were previously that the `GOFIPS` environment variable is set to something at all, but the current plan is to remove this particular environment variable checking support from swarmkit, and just propagate a boolean through.  The caller of `node/node.go` will configure FIPS to be on or off - this should make it easier to test mixed environments in our integration tests.

It's just a big-ish change, so I was going to do it in a separate PR :|